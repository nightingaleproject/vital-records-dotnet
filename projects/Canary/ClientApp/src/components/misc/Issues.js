import React, { Component } from 'react';
import { Icon, Message, Transition } from 'semantic-ui-react';

export class Issues extends Component {
  displayName = Issues.name;

  render() {
    return (
      <React.Fragment>
        {!!this.props.issues && this.props.issues.length > 0 && (
          <div className="inherit-width p-b-50">
            {this.props.issues.map(function(issue, index) {
              return (
                <Transition key={`issue-t-${index}`} transitionOnMount animation="fade" duration={1000}>
                  <div className="inherit-width p-b-10">
                    <Message icon size="large" negative={issue.severity.toLowerCase() === 'error'} warning={issue.severity.toLowerCase() === 'warning'}>
                      <Icon name="exclamation triangle" />
                      <Message.Content>{`${issue.message}`}</Message.Content>
                    </Message>
                  </div>
                </Transition>
              );
            })}
          </div>
        )}
      </React.Fragment>
    );
  }
}
