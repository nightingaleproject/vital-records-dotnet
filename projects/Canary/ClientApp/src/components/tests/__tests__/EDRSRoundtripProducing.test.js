import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import { unmountComponentAtNode } from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import { EDRSRoundtripProducing } from '../EDRSRoundtripProducing';
import { Getter } from '../../misc/Getter';

jest.mock('axios', () => ({
  get: async (uri) => {
    if (uri === '/tests/bfdr-birth/new') {
      return {
        status: 200,
        data: {
          recordId: 0,
          referenceRecord: { recordId: 0 }
        }
      }
    } else {
      return {
        status: 404
      }
    }
  }
}));

const getterSpy = jest.spyOn(Getter.prototype, 'submitPaste');
let container = null;

beforeAll(() => {
  window.API_URL = '';
});

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <EDRSRoundtripProducing params={{ "recordType": "bfdr-birth" }} />
    </MemoryRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should show the success message when issues is defined but empty', async () => {
  getterSpy.mockImplementationOnce(() => {
    const self = getterSpy.mock.instances[0];
    self.props.updateRecord('', []);
  });
  const component = render(
    <MemoryRouter>
      <EDRSRoundtripProducing params={{ "recordType": "bfdr-birth" }} />
    </MemoryRouter>
    , container);
  const user = userEvent.setup();
  const submitButton = await component.findByText('Submit');
  const textArea = await component.findByRole('textbox');
  textArea.focus();
  await userEvent.paste('{ "good": true }');
  fireEvent.click(submitButton);
  const successMessage = await component.findByText('No issues were found!');
  expect(successMessage).toBeDefined();
});

it('should show the problems when there are issues', async () => {
  getterSpy.mockImplementationOnce(() => {
    const self = getterSpy.mock.instances[0];
    self.props.updateRecord('',
      [{ severity: 'error', message: 'Problem description: data no good.' },
      { severity: 'warning', message: 'Also, I am tired.' }]);
  });
  const component = render(
    <MemoryRouter>
      <EDRSRoundtripProducing params={{ "recordType": "bfdr-birth" }} />
    </MemoryRouter>
    , container);
  const user = userEvent.setup();
  const submitButton = await component.findByText('Submit');
  const textArea = await component.findByRole('textbox');
  textArea.focus();
  await userEvent.paste('{ "good": false }');
  fireEvent.click(submitButton);
  const errorMessage = await component.findByText('Problem description: data no good.');
  expect(errorMessage).toBeDefined();
  const warningMessage = await component.findByText('Also, I am tired.');
  expect(warningMessage).toBeDefined();
  await expect(component.findByText('No issues were found!')).rejects.toBeDefined();
});