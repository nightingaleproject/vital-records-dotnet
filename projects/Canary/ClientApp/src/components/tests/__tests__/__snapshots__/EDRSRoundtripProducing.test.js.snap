// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <div
    className="ui grid"
    id="scroll-to"
  >
    <div
      className="row"
    >
      <div
        className="ui breadcrumb"
      >
        <a
          className="section"
          href="/bfdr-birth"
          onClick={[Function]}
        >
          Dashboard
        </a>
        <i
          aria-hidden="true"
          className="right chevron icon divider"
          onClick={[Function]}
        />
        <div
          className="section"
          onClick={[Function]}
        >
           Record Roundtrip (Producing)
        </div>
      </div>
    </div>
    <div
      className="row loader-height"
    >
      <div
        className="ui container"
      >
        <div
          className="ui active transition visible inverted dimmer"
          onClick={[Function]}
        >
          <div
            className="content"
          >
            <div
              className="ui massive text loader"
            >
              Initializing a New Test...
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="p-b-100"
  />,
]
`;
